File: c 풀이/C 프로그래밍 2 D1.c
#include <stdio.h>

int map[2][50 + 10][50 + 10];
int cur;
int n;

void output(void)
{
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			printf("%d ", map[cur][i][j]);
		}
		printf("\n");
	}
}

//cur에서 (cur+1)%2로
void rotation(void)
{
	int after = (cur + 1) % 2;
	int y = n - 1;
	for (int i = 0; i < n; i++) {
		int x = 0;
		for (int j = 0; j < n; j++) {
			map[after][x][y] = map[cur][i][j];
			x += 1;
		}
		y -= 1;
	}
	cur = after;
}

void main(void)
{
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			scanf("%d", &map[cur][i][j]);
		}
	}
	int tmp;
	for (;;) {
		scanf("%d", &tmp);
		if (!tmp) break;
		for (int i = 0; i < (tmp / 90); i++) {
			rotation();
		}
		output();
	}
}

File: python 풀이/C 프로그래밍 2 D1.py
n = int(input())
arr = [ [list(map(int, input().split())) for _ in range(n)], [[0]*n for _ in range(n)] ]
cur = 0

def rotation() :
    global cur
    after = (cur+1) % 2
    y = n-1
    for i in range(n) :
        x = 0
        for j in range(n) :
            arr[after][x][y] = arr[cur][i][j]
            x += 1
        y -= 1
    cur = after

while True :
    tmp = int(input())
    if tmp == 0 : break
    for i in range(tmp//90) :
        rotation()
        for a in arr[cur] :
            print(*a)
